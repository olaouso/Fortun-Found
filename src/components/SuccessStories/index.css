.successStoriesContaier {
  font-family: "Acme";
  margin-bottom: 2em;
}

.successStoriesPosts {
  display: flex;
  align-self: center;
  justify-content: center;
}

.sucessStories {
  color: #c62a88;
  margin-bottom: 0.5em;
  margin-top: 0.5em;
  font-size: 2.5em;
  font-family: "Acme";
}

.weSucceed {
  color: #726a95;
  margin-bottom: 1em;
  margin-top: 1em;
  font-size: 1.25em;
  font-family: "Acme";
}
