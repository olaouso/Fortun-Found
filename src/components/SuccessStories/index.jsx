/* eslint-disable no-unused-expressions */
/* eslint-disable import/extensions */
/* eslint-disable prefer-destructuring */
import React, { useState, useEffect } from "react";
import { useTranslation } from "react-i18next";
import "./index.css";
import { Row, Container } from "react-bootstrap";
import Post from "../Post/index.jsx";
// eslint-disable-next-line import/extensions
import db from "../../firebaseConfig.js";
import i18next from "i18next";

export default function SuccessStories() {
  const [stories, setStories] = useState([]);
  const [currentLanguage, setCurrentLanguage] = useState(i18next.language);
  i18next.on("languageChanged", (lng) => {
    setCurrentLanguage(lng);
  });
  const { t } = useTranslation();

  useEffect(() => {
    const fetchData = async () => {
      const res = await db.collection("donatorPost").get();

      const data = res.docs
        .map((post) => {
          const tmp = post.data();
          const id = post.id;
          return { id, ...tmp };
        })
        .filter((post) => post.lang === currentLanguage);
      setStories(data);
    };
    fetchData();
  }, [currentLanguage]);

  return (
    <Container className="successStoriesContaier">
      <p className="text-center sucessStories">{t("successStories.header")}</p>
      <p className="text-center weSucceed">{t("successStories.paragraph")}</p>
      <Row fluid="md" className="successStoriesPosts">
        {stories
          .map((singleData) => <Post postInfo={singleData} />)
          .slice(0, 3)}
      </Row>
    </Container>
  );
}
