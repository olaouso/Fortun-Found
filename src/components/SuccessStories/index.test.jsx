import React from "react";
import renderer from "react-test-renderer";
import SuccessStories from "./index";

it("renders SuccessStories correctly", () => {
  const tree = renderer.create(<SuccessStories />).toJSON();
  expect(tree).toMatchSnapshot();
});
