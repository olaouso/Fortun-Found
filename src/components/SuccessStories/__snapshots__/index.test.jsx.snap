// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SuccessStories correctly 1`] = `
<div
  className="successStoriesContaier container"
>
  <p
    className="text-center sucessStories"
  >
    successStories.header
  </p>
  <p
    className="text-center weSucceed"
  >
    successStories.paragraph
  </p>
  <div
    className="successStoriesPosts row"
    fluid="md"
  />
</div>
`;
